src/deluge/gui/menu_item/osc/type.h

```diff
@@ -66,6 +68,28 @@ class Type final : public Selection, public FormattedTitle {
 			soundEditor.currentSound->unassignAllVoices();
 		}
 
+		if (newValue == OscType::DX7 && soundEditor.currentSource->dxPatch == nullptr
```

Comment:
My only comment here is that we should probably check that all the pointers are not null before using them